dev/doxygen/pages/building.md

```diff
@@ -0,0 +1,97 @@
+Building LibrePCB {#doc_building}
+=================================
+
+[TOC]
+
+This page is aimed both at developers as well as packagers. If you're a
+packager and have any question regarding the LibrePCB build system, please join
+our [chat or forum](https://librepcb.org/discuss/)!
+
+
+# Building with Qt Creator {#doc_building_qtcreator}
+
+If you want to contribute a new feature or a bugfix to LibrePCB, the best way
+is to use [Qt Creator](https://doc.qt.io/qtcreator/). To get started, simply
+open the `librepcb.pro` file with Qt Creator.
+
+## Configuring the Desktop Kit
+
+When opening a project in Qt Creator for the first time, you need to configure
+the Desktop kit:
+
+![qtcreator_desktopkit](qtcreator_desktopkit.png)
+
+Click on the "Configure Project" button to get started.
+
+## Starting LibrePCB
+
+Select the run configuration `librepcb` and click on the `Run` button:
+
+![qtcreator_run](qtcreator_run.png)
+
+## Parallel Builds
+
+To keep build time as low as possible make sure to set the correct make flags
+to use all available CPU cores to build. See this [stackoverflow
+answer](https://stackoverflow.com/questions/8860712/setting-default-make-options-for-qt-creator).
+
+
+# Building on the Command Line {#doc_building_command_line}
+
+You can also build LibrePCB from the command line using qmake:
+
+    mkdir build && cd build
+    qmake -r ../librepcb.pro
+    make -j8
+
+Then run the `librepcb` library in the `build/output/` directory.
+
+To install LibrePCB on your system, run `make install`.
+
+## Faster Builds with ccache
+
+To speed up subsequent compilation steps, it's recommended to install and
+enable [ccache](https://ccache.dev/) (this will work on Qt 5.9.2 and newer):
```

Comment:
Hmm actually ccache only helps to speed up clean rebuilds, or am I missing something? I mean, usually you don't do clean rebuilds locally, ccache is useful mainly on CI to pass build artifacts between independent build jobs...

Reply:
If you switch between branches, then that may result in recompilation of large parts of LibrePCB, especially if commonly used files, e.g. in librepcbcommon are changed. I'm not sure how much it really helps, but subjectively on my machine it seems that working on LibrePCB is faster with ccache than without.

Reply:
OK I see, but then it would be good to state this more precisely in the documentation :wink: "To speed up subsequent compilation steps, it's recommended..." sounds like compiling is slow by default, which is simply not true. A normal build is probably even slower with ccache enabled (not measured, but ccache of course also needs some processing time).

Reply:
But isn't that exactly what's stated? "To speed up subsequent compilation steps"...

Anyways, I reworded the text to make it clearer.

Reply:
Thanks! :slightly_smiling_face: